controllers/capabilities/tenant_controller.go

```diff
@@ -87,6 +93,12 @@ func (r *TenantReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		return ctrl.Result{}, err
 	}
 
+	// Ignore deleted resources, this can happen when foregroundDeletion is enabled
+	// https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/#foreground-cascading-deletion
+	if tenantR.GetDeletionTimestamp() != nil && !controllerutil.ContainsFinalizer(tenantR, tenantFinalizer) {
+		return ctrl.Result{}, nil
+	}
+
 	if reqLogger.V(1).Enabled() {
```

Comment:
log the object read always (only when debug level). But also when deletion timestamp is added

Reply:
Not sure I follow this comment

Reply:
Hey @eguzki 

We have the intial lgtm on it, I wonder if you could explain to me what exactly is the ask here to move this PR forward since it looks like the last comment left.

Reply:
You fixed already. Before the fix, the object was not logged when deletion timestamp was added to the object.